architecture/2023-12-asyncworker-progressInfo.md

```diff
@@ -0,0 +1,130 @@
+# Title
+
+* **Status**: Pending
+* **Author**: Vinaya Damle (@vinayada1)
+
+## Overview
+
+Currently async worker has no knowledge of the operation progress till the operation is complete. If the operation times out, there is no mechanism in the current design that any information about the operation progress so far can be communicated to the user. As a result, the user only sees the error message that the operation timed out.
```

Comment:
There's the ability to indicate the progress as a percentage, but I agree no real way to provide user-facing messages.

I think the ability to indicate progress is separate from the question of: "What error message do users see when an operation fails?"